### PR TITLE
Season methods branch

### PR DESCRIPTION
**Pull request by sarynm12**
Added the following helper methods: 
-games_won_per_team_for(season_id)
-total_games_per_team_for(season_id) 

Can be used for winningest coach and worst coach 

Pulled down main before pushing this branch up and creating the PR

### CODE REVIEW

**Review by leahriffell — APPROVED** [review]
Nice, Angela! Thanks for doing these! Sorry I misspoke earlier when I said that we both wouldn't need win % by team. I can do that one after this one's merged.

File: lib/stat_tracker.rb [hunk 1]
```diff
@@ -86,6 +86,7 @@ def average_goals_per_game_per_team(exclude_hoa = nil)
     end
   end
 
+
```
Comment (hopesgit) [inline lib/stat_tracker.rb:89]: This is an extra line, not super important

File: test/stat_tracker_test.rb [hunk 2]
```diff
@@ -127,6 +127,7 @@ def test_it_can_get_average_goals_per_game_per_team_hoa
     assert_equal expected_avg_away_goals_p_game_p_team, @@stat_tracker.average_goals_per_game_per_team(exclude)
   end
 
+
```
Comment (hopesgit) [inline test/stat_tracker_test.rb:130]: Extra line here too

File: lib/stat_tracker.rb [hunk 3]
```diff
@@ -97,6 +98,19 @@ def goals_per_game_per_team
...
+      result
+    end
+  end
+
+  def total_games_per_team_for(season_id)
+    @game_teams.reduce(Hash.new(0)) do |result, game_team|
+      result[game_team.team_id] += 1 if game_team.season == season_id
+      result
+    end
+  end
```
Comment (hopesgit) [inline lib/stat_tracker.rb:113]: These look solid, nice job!

**Review by hopesgit — APPROVED** [review]

**Review by AngelaGuardia — APPROVED** [review]
Great job on these Saryn!

**Pull request merged**